### PR TITLE
Allow custom checkbox component

### PR DESCRIPTION
**Pull request by azhangstrata**
## Description
A few sentences describing the overall goals of the pull request's commits.

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[X ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: packages/react-data-grid/src/ReactDataGrid.js [hunk 1]
```diff
@@ -112,7 +112,9 @@ const ReactDataGrid = React.createClass({
     /* called before cell is set active, returns a boolean to determine whether cell is editable */
     overScan: React.PropTypes.object,
     onDeleteSubRow: React.PropTypes.func,
-    onAddSubRow: React.PropTypes.func
+    onAddSubRow: React.PropTypes.func,
+    selectionRenderer: React.PropTypes.object,
```
Comment (diogofcunha) [inline packages/react-data-grid/src/ReactDataGrid.js]: Hmm, this is not an object is it? It's a func

File: packages/react-data-grid/src/ReactDataGrid.js [hunk 2]
```diff
@@ -843,12 +845,14 @@ const ReactDataGrid = React.createClass({
...
     let unshiftedCols = {};
     if (this.props.rowActionsCell || (props.enableRowSelect && !this.props.rowSelection) || (props.rowSelection && props.rowSelection.showCheckbox !== false)) {
-      let headerRenderer = props.enableRowSelect === 'single' ? null :
-      <div className="react-grid-checkbox-container checkbox-align">
-        <input className="react-grid-checkbox" type="checkbox" name="select-all-checkbox" id="select-all-checkbox" ref={grid => this.selectAllCheckbox = grid} onChange={this.handleCheckboxChange} />
-        <label htmlFor="select-all-checkbox" className="react-grid-checkbox-label"></label>
-      </div>;
-      let Formatter = this.props.rowActionsCell ? this.props.rowActionsCell : CheckboxEditor;
+      const SelectAll = this.props.selectAllRenderer;
+      const SelectAllRenderer = SelectAll ?  <SelectAll onChange={this.handleCheckboxChange} /> : (
```
Comment (diogofcunha) [inline packages/react-data-grid/src/ReactDataGrid.js]: Wouldn't it much easier if you just:
- Create a component for the `SelectAll` in a new file.
- Set that component as a default prop for `SelectAllRenderer`?

File: packages/react-data-grid/src/ReactDataGrid.js [hunk 3]
```diff
@@ -843,12 +845,14 @@ const ReactDataGrid = React.createClass({
...
+        <div className="react-grid-checkbox-container checkbox-align">
+          <input className="react-grid-checkbox" type="checkbox" name="select-all-checkbox" id="select-all-checkbox" ref={grid => this.selectAllCheckbox = grid} onChange={this.handleCheckboxChange} />
+          <label htmlFor="select-all-checkbox" className="react-grid-checkbox-label"></label>
+        </div>);
+      let headerRenderer = props.enableRowSelect === 'single' ? null : SelectAllRenderer;
+      let Formatter = this.props.rowActionsCell ? this.props.rowActionsCell : this.props.selectionRenderer || CheckboxEditor;
```
Comment (diogofcunha) [inline packages/react-data-grid/src/ReactDataGrid.js]: Hmm, what is rowActionsCell? Have you dig it out? Isn't it to be used as a renderer?

File: packages/react-data-grid/src/formatters/SelectAll.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    <div className="react-grid-checkbox-container checkbox-align">
+      <input className="react-grid-checkbox" type="checkbox" name="select-all-checkbox" id="select-all-checkbox" ref={props.inputRef} onChange={props.onChange} />
+      <label htmlFor="select-all-checkbox" className="react-grid-checkbox-label"></label>
+    </div>
+  );
+};
+
+SelectAll.propTypes = {
+  onChange: PropTypes.func,
+  inputRef: PropTypes.element
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/formatters/SelectAll.js]: `inputRef` should be a func?

**Comment by richardware** [comment]
@azhangstrata can you please run a catch-up and resolve the conflict above?

Do we still even need this PR?

File: packages/react-data-grid-addons/src/__tests__/Grid.spec.js [hunk 5]
```diff
@@ -233,10 +233,10 @@ describe('Grid', function() {
         let checkboxWrapper = document.createElement('div');
         checkboxWrapper.innerHTML = '<input type="checkbox" value="value" checked="true" />';
         this.checkbox = checkboxWrapper.querySelector('input');
-
-        this.headerCheckbox = this.selectRowCol.headerRenderer.props.children[0];
+        const SelectAll = this.selectRowCol.headerRenderer;
+        this.selectAllWrapper = mount(SelectAll);
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/__tests__/Grid.spec.js]: Does `shallow` work in this case?

**Pull request merged**